controllers/movies.ts

```diff
@@ -0,0 +1,302 @@
+          actor: `%${actor}%`,
+        });
+      }
+      if (imdbSort) {
+        const imdbSortDirection =
+          imdbSort.toUpperCase() === "ASC" ? "ASC" : "DESC";
+        query = query.orderBy("movie.imdbRating", imdbSortDirection);
+      }
+
+      if (yearSort) {
+        const yearSortDirection =
+          yearSort.toUpperCase() === "ASC" ? "ASC" : "DESC";
+        query = query.orderBy("movie.Year", yearSortDirection);
+      }
+      let movies;
+
+      if (imdbSort) {
+        movies = await query
+          .select(["movie.Title", "movie.imdbRating"])
+          .getMany();
+      } else {
+        movies = await query.getMany();
+      }
+      return movies;
+    } catch (err: any) {
+      throw new Error("Movies not found");
+    }
+  }
+
+  async getMovieByImdID(imdb: any) {
+    try {
+      const movie = await Movie.findOne({
+        where: {
+          imdbID: imdb,
+        },
+      });
+      return movie;
+    } catch (err: any) {
+      throw new Error("Movie not found");
+    }
+  }
+
+  async deleteMovie(id: any) {
+    try {
+      const movie = (await Movie.findOne({
+        where: {
+          Id: id,
+        },
+      })) as Movie;
+      await Movie.remove(movie);
+      return movie;
+    } catch (err: any) {
+      throw new Error("Movie not found");
+    }
+  }
+
+  async addMovie(params: MovieParams) {
+    const {
+      Title,
+      Year,
+      Rated,
+      Released,
+      Runtime,
+      Genre,
+      Director,
+      Writer,
+      Actors,
+      Plot,
+      Language,
+      Country,
+      Awards,
+      Poster,
+      Metascore,
+      imdbRating,
+      imdbVotes,
+      imdbID,
+      Type,
+      Response,
+      Images,
+      ComingSoon,
+    } = params;
+
+    try {
+      const movie = await Movie.create({
+        Title: Title,
+        Year: Year,
+        Rated: Rated,
+        Released: Released,
+        Runtime: Runtime,
+        Genre: Genre,
+        Director: Director,
+        Writer: Writer,
+        Actors: Actors,
+        Plot: Plot,
+        Language: Language,
+        Country: Country,
+        Awards: Awards,
+        Poster: Poster,
+        Metascore: Metascore,
+        imdbRating: imdbRating,
+        imdbVotes: imdbVotes,
+        imdbID: imdbID,
+        Type: Type,
+        Response: Response,
+        Images: Images,
+        ComingSoon: ComingSoon,
+      });
+
+      await movie.save();
+      return movie;
+    } catch (err: any) {
+      throw new Error("Movie not found");
+    }
+  }
+
+  async editMovie(params: MovieParams, id: any) {
+    const {
+      Title,
+      Year,
+      Rated,
+      Released,
+      Runtime,
+      Genre,
+      Director,
+      Writer,
+      Actors,
+      Plot,
+      Language,
+      Country,
+      Awards,
+      Poster,
+      Metascore,
+      imdbRating,
+      imdbVotes,
+      imdbID,
+      Type,
+      Response,
+      Images,
+      ComingSoon,
+    } = params;
+
```

Comment:
Instead of writing all of the fields here, you can just pass the whole object params to the save method, if we want to store all of the sent fields. Before doing any action into the database please make sure to validate the parameters, maybe try using Joi for validation of the fields.